src/Apps/Order/Routes/Shipping2/shippingUtils.ts

```diff
@@ -0,0 +1,229 @@
+  >["edges"]
+>
+
+export const getDefaultUserAddress = (addressList: SavedAddressType[]) => {
+  const items = compact(addressList)
+
+  if (!items || items.length == 0) {
+    return
+  }
+
+  return items.find(node => node.isDefault) || items[0]
+}
+
+const matchAddressFields = (...addressPair: [object, object]) => {
+  const [a1, a2] = addressPair.map(a => addressWithFallbackValues(a))
+  return (
+    a1.addressLine1 === a2.addressLine1 &&
+    a1.addressLine2 === a2.addressLine2 &&
+    a1.city === a2.city &&
+    a1.country === a2.country &&
+    a1.name === a2.name &&
+    a1.phoneNumber === a2.phoneNumber &&
+    a1.postalCode === a2.postalCode &&
+    a1.region === a2.region
+  )
+}
+
+const getSavedFulfillmentData = (
+  order: ShippingProps["order"],
+  me: ShippingProps["me"]
+): {
+  fulfillmentType: FulfillmentType
+  isArtsyShipping: boolean
+  fulfillmentDetails: FulfillmentValues["attributes"]
+  selectedSavedAddressId: string | null
+} | null => {
+  if (
+    !order.requestedFulfillment ||
+    Object.keys(order.requestedFulfillment).length === 0
+  ) {
+    return null
+  }
+
+  const requestedFulfillmentType = order.requestedFulfillment.__typename
+  if (requestedFulfillmentType === "CommercePickup") {
+    const phoneNumber = order.requestedFulfillment.phoneNumber!
+    return {
+      fulfillmentType: FulfillmentType.PICKUP,
+      isArtsyShipping: false,
+      // TODO: [When things are working again]
+      // figure out what `name` is used for w/ pickup, where to get it from
+      fulfillmentDetails: { phoneNumber } as FulfillmentValues["attributes"],
+      selectedSavedAddressId: null,
+    }
+  }
+  const fulfillmentDetails: ShippingAddressFormValues = addressWithFallbackValues(
+    order.requestedFulfillment
+  )
+
+  const addressList =
+    extractNodes(me?.addressConnection) ??
+    ([] as SavedAddressType[]).filter(a => !!a)
+
+  // we don't store the address id on exchange orders, so we need to match every field
+  const selectedSavedAddressId =
+    addressList.find(node => matchAddressFields(node, fulfillmentDetails))
+      ?.internalID || null
+
+  if (requestedFulfillmentType === "CommerceShipArta") {
+    return {
+      fulfillmentType: FulfillmentType.SHIP,
+      isArtsyShipping: true,
+      fulfillmentDetails,
+      selectedSavedAddressId,
+    }
+  }
+  if (requestedFulfillmentType === "CommerceShip") {
+    return {
+      fulfillmentType: FulfillmentType.SHIP,
+      isArtsyShipping: false,
+      fulfillmentDetails,
+      selectedSavedAddressId,
+    }
+  }
+  // Should never happen. Log it?
+  return null
+}
+
+// Interface to separate local values from what we want to expose
+// in the react context.
+interface OrderData extends ShippingContextProps {
+  shippingQuotesEdges: ShippingQuotesConnectionEdges
+  selectedShippingQuoteId?: string
+}
+
+// Compute and memoize data from the saved order.
+export const useLoadComputedData = (props: ShippingProps): OrderData => {
```

Comment:
The structure of Shipping2/shippingUtils.ts is similar to the existing shippingUtils.ts, but with much fewer exports. Most things about inferring the state of the order are contained in this one giant hook. This is used to always reflect current order state from the relay-provided order prop instead of maintaining our own copies in the shipping route, and was one of the main challenges i found in the refactor.